app/api/companies.py

```diff
@@ -138,24 +140,41 @@ def get_company(id):
 def add_company():
     data = request.get_json() or {}
 
-    # Validation
-    if 'company_id' in data:
-        return bad_request("Create company cannot include company_id. For modifying existing companies please use the PUT method")
-    if 'company_name' not in data:
-        return bad_request("Must include company_name field")
-    if Companies.query.filter_by(company_name=data['company_name']).first():
-        return bad_request("A company with that name already exists, please use another name")
-
-    company = Companies()
-    # TODO: Finalize the from_dict method
-    company.from_dict_new(data)
-    db.session.add(company)
-    db.session.commit()
-
-    response = jsonify(company.to_dict())
+    try:
+        result = CompaniesValidationSchema().load(data)
```

Comment:
Love it - this is a much more `extendible` approach where we can add more rules to the Schema as we learn more about the requirements of the app, but the code here can stay the same 🔥